tests/test_model.py

```diff
@@ -18,3 +24,47 @@ def test_model_building(input_dims, embedding_dims, output_dims):
     assert model.conv1.in_channels == input_dims
     assert model.linear.in_features == embedding_dims
     assert model.linear.out_features == output_dims
+
+
+@pytest.mark.parametrize(
+    "bbox_size, model",
+    [
+        ((2, 2), lambda x: torch.sum(x)),
+        ((3, 3), lambda x: torch.mean(x)),
+    ],
+)
+class TestFeatureExtractor:
+    @pytest.fixture
+    def vars(self, bbox_size, model, default_rng):
+        image, graph = random_image_and_graph(default_rng)
+        return {
+            "extractor": FeatureExtractor(
+                bbox_size=bbox_size,
+                model=model,
+                transforms=lambda x: x,
+                augmentations=lambda x: x,
+            ),
+            "image": torch.tensor(image.astype("float32")),
+            "graph": graph,
+        }
+
+    def test_feature_extractor_forward(self, bbox_size, model, vars):
+        image_out, target_out = vars["extractor"](
+            vars["image"], {"graph": vars["graph"]}
+        )
+        graph_out = target_out["graph"]
+
+        assert torch.equal(vars["image"], image_out)
+
+        for node in graph_out.nodes.data():
+            x, y = int(node[1][GraphAttrs.NODE_X]), int(
+                node[1][GraphAttrs.NODE_Y]
+            )
+            features = node[1][GraphAttrs.NODE_FEATURES]
```

Comment:
does the test check also the feature size is the expected? like how the one for resnet152 would be 2048, etc

Reply:
Not specifically for resent152, no. Would this be useful? My idea was since it's meant to be model agnostic, there shouldn't be explicit checks for each individual model here, but rather, if we write config tests, or tests for the whole pipeline, then it should check that the extractor output matches the GCN input in this way.

Reply:
yes, I meant it in an agnostic way, not just specifically for resnet152! something to check that the dimensions are correct and what was expected, but that works out better

Reply:
Makes sense! Yeah I'll make sure to add those in the pipeline tests